FreeRTOS/FreeRTOS-Cellular-Interface
# Update cellular platform header APIs

Description
-----------
Cellular interface makes use of the platform header to abstract FreeRTOS dependency.

In this PR:
* Move the following FreeRTOS functions and data structure to platform
```
    #define PlatformQueueHandle_t    QueueHandle_t
    #define PlatformEventBits_t      EventBits_t
    #define PlatformBaseType_t       BaseType_t
    #define PlatformTickType_t       TickType_t
    #define platformTRUE             pdTRUE
    #define platformFALSE            pdFALSE
    #define platformPASS             pdPASS
    #define platformFAIL             pdFAIL
    #define platformMAX_DELAY        portMAX_DELAY
    #define PlatformQueue_Create     xQueueCreate
    #define PlatformQueue_Send       xQueueSend
    #define PlatformQueue_Receive    xQueueReceive
    #define PlatformQueue_Delete     vQueueDelete
```
* Add `CELLULAR_CONFIG_ASSERT` config. Default value is `configASSERT`
* Remove critical section dependency. The common layer function is not thread safe.

Test Steps
-----------
N/A

Checklist:
----------


- [] ~~I have modified and/or added unit-tests to cover the code changes in this Pull Request.~~

Related Issue
-----------

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.
